### DISCUSSION TITLE
why? my decompress is long time

### DISCUSSION THREAD

**Discussion by jiangjiangjiangjiangjiangjiangjiang** [klauspost/compress#560] [General]
zstd file size: 222M
take time of zstd decompress: 4s

my code:
```go
func ZstdDecompressFile(src, dst string) error {
    in_file, err := os.Open(src)
    if err != nil {
        return err
    }
    defer in_file.Close()
    out_file, err := os.Create(dst)
    if err != nil {
        return err
    }
    defer out_file.Close()

    zst_reader, err := zstd.NewReader(in_file)
    if err != nil {
        return err
    }

    _, err = io.Copy(out_file, zst_reader)
    if err != nil {
        zst_reader.Close()
        return err
    }

    return nil
}
```

**Comment by klauspost** [comment]
Slow disk?

How long does it take to copy `ioutil.Discard`?

I am really just guessing. It could also be that your output file is huge. No idea really.

Provide more info.

**Reply by jiangjiangjiangjiangjiangjiangjiang** [reply]
the decompressed file(tar file) size is 1G
ioutil.Discard's  time about 2s
so, the decompress speed is 1G/2s = 500M/s now.

**Reply by klauspost** [reply]
That is the expected speed.

Assembly is currently being finalized so the next release will likely be faster.

**Reply by jiangjiangjiangjiangjiangjiangjiang** [reply]
wow, it is magic...
